olive/engine/cloud_cache_helper.py

```diff
@@ -46,40 +45,61 @@ def __init__(
             account_url=account_url, container_name=self.container_name, credential=credential
         )
 
-        self.cache_dir = cache_dir
-        self.output_model_path = Path(cache_dir) / "model"
+        self.model_path_map = "model_path.json"
         self.input_model_config = input_model_config
 
-    def update_model_config(self, cloud_model_path, model_config, input_model_hash):
+    def update_model_config(self, cloud_model_path, adapter_path, model_config, input_model_hash, output_model_path):
         logger.info("Updating model config with cloud model path: %s", cloud_model_path)
         model_directory_prefix = f"{input_model_hash}/model"
         blob_list = self.container_client.list_blobs(name_starts_with=model_directory_prefix)
+        self.download_blob_list(blob_list, model_directory_prefix, output_model_path)
 
+        adapter_directory_prefix = f"{input_model_hash}/adapter"
+        blob_list = self.container_client.list_blobs(name_starts_with=adapter_directory_prefix)
+        self.download_blob_list(blob_list, adapter_directory_prefix, output_model_path, "adapter")
+
+        if model_config.type.lower() == "hfmodel" and is_hf_repo_exist(cloud_model_path):
+            model_config.config["model_path"] = cloud_model_path
+        else:
+            model_config.config["model_path"] = str(output_model_path / cloud_model_path)
+
+        if model_config.type.lower() == "hfmodel" and adapter_path:
+            model_config.config["adapter_path"] = str(output_model_path / adapter_path)
+
+        return model_config
+
+    def download_blob_list(self, blob_list, directory_prefix, output_model_path, prefix: str = None):
         for blob in blob_list:
             blob_client = self.container_client.get_blob_client(blob)
-            local_file_path = self.output_model_path / blob.name[len(model_directory_prefix) + 1 :]
+            local_file_path = (
+                output_model_path / prefix / blob.name[len(directory_prefix) + 1 :]
```

Comment:
instead of doing string slice, maybe you could do something like `Path(blob.name).name`?

Reply:
`Path(blob.name).name` will lose the subfolder structure of this blob. For example if I have `prefix/subfolder/file.name`, `Path(blob.name).name` will only get `file.name`.

Reply:
Oh I see.